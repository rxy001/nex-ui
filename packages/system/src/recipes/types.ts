/* eslint-disable no-use-before-define */
import type { StyleObject, Dictionary } from '../types'

type BooleanMap<T> = T extends 'true' | 'false' ? boolean : T

export type SlotGroups = Dictionary<StyleObject>

export type BaseVariantGroups = Dictionary<Dictionary<StyleObject>>

export type SlotVariantGroups<S extends SlotGroups = SlotGroups> = Dictionary<
  Dictionary<Partial<Record<keyof S, StyleObject>>>
>

type CompoundVariantsSelection<
  V extends BaseVariantGroups | SlotVariantGroups,
> = {
  [K in keyof V]?: string extends K
    ? unknown
    : BooleanMap<keyof V[K]> | BooleanMap<keyof V[K]>[]
}

type VariantSelection<V extends BaseVariantGroups | SlotVariantGroups> = {
  [K in keyof V]?: string extends K ? unknown : BooleanMap<keyof V[K]>
}

interface RuntimeFn<V extends Record<string, any>, R> {
  (variants?: VariantSelection<V>): R
}

export type CombineVariants<
  V,
  E extends RecipeRuntimeFn | SlotRecipeRuntimeFn | undefined,
> = E extends RecipeRuntimeFn | SlotRecipeRuntimeFn
  ? V & E['__config']['variants']
  : V

export type UniteVariants<
  V,
  T extends RecipeRuntimeFn | SlotRecipeRuntimeFn | undefined,
  R,
  Key = T extends RecipeRuntimeFn | SlotRecipeRuntimeFn
    ? T['__config']['variants']
    : undefined,
> =
  | V
  | {
      [K in keyof Key]?: {
        [P in keyof Key[K]]?: R
      }
    }

export type RecipeConfig<
  V extends BaseVariantGroups = BaseVariantGroups,
  E extends RecipeRuntimeFn | undefined = undefined,
> = {
  base?: StyleObject
  variants?: UniteVariants<V, E, StyleObject>
  compoundVariants?: Array<
    CompoundVariantsSelection<CombineVariants<V, E>> & {
      css?: StyleObject
    }
  >
  defaultVariants?: VariantSelection<CombineVariants<V, E>>
}

export interface RecipeRuntimeFn<
  V extends BaseVariantGroups = BaseVariantGroups,
> extends RuntimeFn<V, StyleObject> {
  __recipe: true
  __config: RecipeConfig<V>
  variants: (keyof V)[]
}

export type UniteSlots<
  V,
  E extends SlotRecipeRuntimeFn | undefined,
  Key = E extends SlotRecipeRuntimeFn ? E['__config']['slots'] : undefined,
> =
  | V
  | {
      [K in keyof Key]: StyleObject
    }

type FilterIndexKey<T> = {
  [K in keyof T as string extends K ? never : K]: T[K]
}

export type CombineSlots<
  S,
  E extends SlotRecipeRuntimeFn | undefined,
  Key = E extends SlotRecipeRuntimeFn
    ? E['__config']['slots']
    : NonNullable<unknown>,
> = FilterIndexKey<
  S & {
    [K in keyof Key]: StyleObject
  }
>

export type SlotRecipeConfig<
  S extends SlotGroups = SlotGroups,
  E extends SlotRecipeRuntimeFn | undefined = undefined,
  V extends SlotVariantGroups<CombineSlots<S, E>> = SlotVariantGroups<
    CombineSlots<S, E>
  >,
> = {
  slots?: Partial<UniteSlots<S, E>>
  variants?: UniteVariants<
    V,
    E,
    {
      [K in keyof CombineSlots<S, E>]?: StyleObject
    }
  >
  compoundVariants?: Array<
    CompoundVariantsSelection<CombineVariants<V, E>> & {
      css?: {
        [K in keyof CombineSlots<S, E>]?: StyleObject
      }
    }
  >
  defaultVariants?: VariantSelection<CombineVariants<V, E>>
}

export interface SlotRecipeRuntimeFn<
  S extends SlotGroups = SlotGroups,
  V extends SlotVariantGroups<S> = SlotVariantGroups<S>,
> extends RuntimeFn<V, Record<keyof S, StyleObject>> {
  __slotRecipe: true
  __config: SlotRecipeConfig<S, undefined, V>
  variants: (keyof V)[]
  slots: (keyof S)[]
}

export type RecipeVariants<
  RecipeFn extends RecipeRuntimeFn | SlotRecipeRuntimeFn,
  V = Parameters<RecipeFn>[0],
> = Exclude<
  {
    [Key in keyof V]: V[Key]
  } & NonNullable<unknown>,
  undefined
>
