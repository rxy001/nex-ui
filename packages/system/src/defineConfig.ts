import type { SemanticTokensDefinition, TokensDefinition } from './tokens'
import type { SystemConfig } from './system'

export const defineConfig = (config: SystemConfig) => config
defineConfig.selectors = (v: SystemConfig['selectors']) => v
defineConfig.aliases = (v: SystemConfig['aliases']) => v
defineConfig.scales = (v: SystemConfig['scales']) => v
defineConfig.breakpoints = (v: SystemConfig['breakpoints']) => v

export const defineTokens = (v: TokensDefinition) => v
defineTokens.colors = (v: TokensDefinition['colors']) => v
defineTokens.borders = (v: TokensDefinition['borders']) => v
defineTokens.lineHeights = (v: TokensDefinition['lineHeights']) => v
defineTokens.fontFamilies = (v: TokensDefinition['fontFamilies']) => v
defineTokens.fontSizes = (v: TokensDefinition['fontSizes']) => v
defineTokens.fontWeights = (v: TokensDefinition['fontWeights']) => v
defineTokens.sizes = (v: TokensDefinition['sizes']) => v
defineTokens.spaceing = (v: TokensDefinition['spacing']) => v
defineTokens.radii = (v: TokensDefinition['radii']) => v

export const defineSemanticTokens = (v: SemanticTokensDefinition) => v
defineSemanticTokens.colors = (v: SemanticTokensDefinition['colors']) => v
defineSemanticTokens.borders = (v: SemanticTokensDefinition['borders']) => v
defineSemanticTokens.lineHeights = (
  v: SemanticTokensDefinition['lineHeights'],
) => v
defineSemanticTokens.fontFamilies = (
  v: SemanticTokensDefinition['fontFamilies'],
) => v
defineSemanticTokens.fontSizes = (v: SemanticTokensDefinition['fontSizes']) => v
defineSemanticTokens.fontWeights = (
  v: SemanticTokensDefinition['fontWeights'],
) => v
defineSemanticTokens.sizes = (v: SemanticTokensDefinition['sizes']) => v
defineSemanticTokens.spaceing = (v: SemanticTokensDefinition['spacing']) => v
defineSemanticTokens.radii = (v: SemanticTokensDefinition['radii']) => v
