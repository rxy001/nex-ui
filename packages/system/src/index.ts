export * from '@emotion/react'

export * from './Provider'

export { createSystem } from './system'

export { defineStyles } from './styles'
export type {
  StylesFn,
  BaseStylesDefinition,
  SlotStylesDefinition,
  StylesDefinition,
} from './styles'

export { defineConfig } from './defineConfig'

export type {
  CSSInterpolation,
  CSSProperties,
  SystemDefinition,
  CSSPropertiesOverrides,
  StyleObject,
  NormalizeFn,
} from './types'

export type { AliasesDefinition } from './aliases'
export type { ScalesDefinition } from './scales'
export type {
  BordersDefinition,
  ColorsDefinition,
  LineHeightsDefinition,
  FontFamiliesDefinition,
  FontSizesDefinition,
  FontWeightsDefinition,
  SizesDefinition,
  SpacingDefinition,
  RadiiDefinition,
  SemanticDefinition,
} from './tokens'
