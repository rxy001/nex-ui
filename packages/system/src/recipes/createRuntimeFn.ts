import { forEach, merge, isArray } from '@nex-ui/utils'
import { memoizeFn } from '../utils'

export function createRuntimeFn(options?: any) {
  const {
    mainStyles,
    variants = {},
    compoundVariants = [],
    defaultVariants = {},
  } = options ?? {}

  function shouldApplyCompound(compoundCheck: any, selections: any) {
    for (const key in compoundCheck) {
      if (Object.prototype.hasOwnProperty.call(compoundCheck, key)) {
        const variantValue = selections[key] ?? defaultVariants?.[key]
        if (
          !(
            (isArray(compoundCheck[key]) &&
              compoundCheck[key]?.includes(variantValue)) ||
            compoundCheck[key] === variantValue
          )
        ) {
          return false
        }
      }
    }

    return true
  }

  function splitVariantProps(props: any) {
    const variantKeys = Object.keys(variants)
    const result: any = {}

    forEach(variantKeys, (key: any) => {
      if (props[key] !== undefined) result[key] = props[key]
    })

    return result
  }

  function runtimeFn(variantsProps: any = {}) {
    let mergedStyles = {
      ...mainStyles,
    }

    const selections = {
      ...defaultVariants,
      ...splitVariantProps(variantsProps),
    }

    forEach(selections, (variantSelection: any, variantName: any) => {
      if (variantSelection !== null) {
        let selection = variantSelection as string | boolean

        if (typeof selection === 'boolean') {
          selection = selection === true ? 'true' : 'false'
        }
        mergedStyles = merge(
          {},
          mergedStyles,
          variants?.[variantName]?.[selection],
        )
      }
    })

    forEach(compoundVariants, (compoundVariant: any) => {
      const { css: compoundVariantCSS, ...compoundCheck } = compoundVariant

      if (shouldApplyCompound(compoundCheck as any, selections)) {
        mergedStyles = merge({}, mergedStyles, compoundVariantCSS)
      }
    })

    return mergedStyles
  }

  return memoizeFn(runtimeFn)
}
