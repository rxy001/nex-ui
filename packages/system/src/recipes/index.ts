export { defineRecipe } from './defineRecipe'
export { defineSlotRecipe } from './defineSlotRecipe'

export type {
  RecipeConfig,
  RecipeRuntimeFn,
  SlotRecipeConfig,
  SlotRecipeRuntimeFn,
  RecipeVariants,
} from './types'
