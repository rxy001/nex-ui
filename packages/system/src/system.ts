import { forEach, isPlainObject, isString } from '@nex-ui/utils'
import { pathToName } from './utils'
import type { SystemConfig } from './types'
import { createStylesFn } from './styles'
import { createTokens } from './tokens'
import { createScales } from './scales'
import { createAliases } from './aliases'

export const createSystem = (config: SystemConfig) => {
  const { cssVarsPrefix = 'system', scales, aliases, ...tokens } = config

  const { getToken, getCssVars } = createTokens({
    tokens,
    prefix: cssVarsPrefix,
  })

  const { getPropScale } = createScales({ scales })

  const { getProperties } = createAliases({ aliases })

  const normalize = <T extends Record<string, any>>(
    style: T,
    specifiedColorPalette?: string,
  ) => {
    const result: any = Array.isArray(style) ? [] : {}

    forEach(style, (value: any, key: string) => {
      const properties = getProperties(key) ?? [key]
      forEach(properties, (property: string) => {
        const category = getPropScale(property)
        if (category) {
          let newValue = value
          switch (category) {
            case 'colors':
              if (isString(newValue)) {
                newValue = specifiedColorPalette
                  ? newValue.replace(
                      /^colorPalette(?=\.)/,
                      specifiedColorPalette,
                    )
                  : newValue
              }

              break
            default:
              break
          }

          const token = getToken(pathToName([category, newValue]))
          result[property] = token?.value ?? value
        } else if (isPlainObject(value) || Array.isArray(value)) {
          // cssobject
          result[property] = normalize(value as any, specifiedColorPalette)
        } else {
          // 标准 css 值
          result[property] = value
        }
      })
    })

    return result as T
  }

  const styles = createStylesFn({
    normalize,
  })

  return {
    styles,
    normalize,
    globalCssVars: {
      ':root': getCssVars(),
    },
  }
}
