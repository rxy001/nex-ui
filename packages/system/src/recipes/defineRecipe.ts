import { merge } from '@nex-ui/utils'
import { createRuntimeFn } from './createRuntimeFn'
import { memoizeFn } from '../utils'
import type {
  RecipeConfig,
  BaseVariantGroups,
  RecipeRuntimeFn,
  CombineVariants,
} from './types'

function defineRecipeImpl<V extends BaseVariantGroups>(
  config: RecipeConfig<V>,
): RecipeRuntimeFn<V>
function defineRecipeImpl<
  V extends BaseVariantGroups,
  E extends RecipeRuntimeFn,
>(extend: E, config: RecipeConfig<V, E>): RecipeRuntimeFn<CombineVariants<V, E>>
function defineRecipeImpl<
  V extends BaseVariantGroups,
  E extends RecipeRuntimeFn,
>(extendOrConfig: E | RecipeConfig<V>, maybeConfig?: RecipeConfig<V, E>) {
  let config: RecipeConfig<any> = maybeConfig || extendOrConfig

  if (
    (extendOrConfig as E)?.__recipe === true &&
    (extendOrConfig as E)?.__config
  ) {
    config = merge({}, (extendOrConfig as E).__config, config)
  }

  const { base, ...other } = config
  const runtimeFn = createRuntimeFn({
    mainStyles: base,
    ...other,
  }) as any

  runtimeFn.__recipe = true
  runtimeFn.__config = config
  runtimeFn.variants = config.variants ? Object.keys(config.variants) : []

  return runtimeFn
}

export const defineRecipe = memoizeFn(defineRecipeImpl)
