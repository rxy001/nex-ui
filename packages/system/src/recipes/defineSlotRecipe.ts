import { isArray, __DEV__ } from '@nex-ui/utils'
import { createRuntimeFn } from './createRuntimeFn'
import { mergeRecipeConfigs } from '../utils'
import type {
  SlotGroups,
  SlotRecipeConfig,
  SlotVariantGroups,
  SlotRecipeRuntimeFn,
  MergeVariants,
  MergeSlots,
} from './types'

export function defineSlotRecipe<
  Slots extends SlotGroups,
  RuntimeFn extends SlotRecipeRuntimeFn | undefined = undefined,
  Variants extends SlotVariantGroups<
    MergeSlots<Slots, RuntimeFn>
  > = SlotVariantGroups<MergeSlots<Slots, RuntimeFn>>,
>(
  config: SlotRecipeConfig<Slots, RuntimeFn, Variants>,
): SlotRecipeRuntimeFn<
  MergeSlots<Slots, RuntimeFn>,
  MergeVariants<Variants, RuntimeFn>
> {
  if (__DEV__) {
    if (config.compoundVariants && !isArray(config.compoundVariants)) {
      throw new TypeError(
        `[Nex UI] system: The "compoundVariants" prop must be an array. Received: ${typeof config.compoundVariants}`,
      )
    }
  }

  // eslint-disable-next-line prefer-const
  let { extend, ...other } = config

  if (extend && extend.__slotRecipe === true && extend.__config) {
    other = mergeRecipeConfigs(extend.__config, other)
  }

  const { slots, ...o } = other

  const runtimeFn = createRuntimeFn({
    mainStyles: slots,
    ...o,
  }) as any

  runtimeFn.__slotRecipe = true
  runtimeFn.__config = other
  runtimeFn.variants = other.variants ? Object.keys(other.variants) : []
  runtimeFn.slots = other.slots ? Object.keys(other.slots) : []

  return runtimeFn
}
