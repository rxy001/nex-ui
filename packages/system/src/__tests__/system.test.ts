import { describe, it, expect } from '@jest/globals'
import { defineRecipe } from '../recipes'
import { createSystem } from '../system'
import { defineConfig } from '../defineConfig'

const SELECTOR_HORVER = '&:not(:disabled):not([data-disabled=true]):hover'
const CSS_VARS_PREFIX = 'test'

const sysConfig = defineConfig({
  cssVarsPrefix: CSS_VARS_PREFIX,
  aliases: {
    w: 'width',
    h: 'height',
    px: ['paddingLeft', 'paddingRight'],
    fs: 'fontSize',
    fw: 'fontWeight',
    lh: 'lineHeight',
  },
  breakpoints: { sm: '500px', md: '700px', lg: '900px' },
  tokens: {
    colors: {
      white: '#f1f1f1',
      blue: {
        100: '#dbeafe',
        200: '#bfdbfe',
        300: '#a3cfff',
      },
    },
    borders: { xs: '0.5px', sm: '1px' },
    radii: { sm: '4px', md: '6px', lg: '8px' },
    sizes: {
      1: '32px',
      2: '40px',
      3: '48px',
    },
    spaces: {
      1: '4px',
      2: '8px',
      3: '12px',
    },
    fontSizes: {
      sm: '12px',
      md: '14px',
      lg: '16px',
    },
    fontWeights: {
      normal: 400,
      medium: 500,
      semibold: 600,
    },
    lineHeights: {
      short: 1.375,
      base: 1.5,
      tall: 1.625,
    },
  },
  scales: {
    color: 'colors',
    borderColor: 'colors',
    backgroundColor: 'colors',

    fontSize: 'fontSizes',
    borderWidth: 'borders',

    width: 'sizes',
    height: 'sizes',

    fontWeight: 'fontWeights',
    lineHeight: 'lineHeights',
    fontFamily: 'fontFamilies',

    gap: 'spaces',

    padding: 'spaces',
    paddingTop: 'spaces',
    paddingBottom: 'spaces',
    paddingLeft: 'spaces',
    paddingRight: 'spaces',

    marginRight: 'spaces',
    marginTop: 'spaces',
    marginLeft: 'spaces',
    marginBottom: 'spaces',
    margin: 'spaces',

    borderRadius: 'radii',
    borderTopRightRadius: 'radii',
    borderTopLeftRadius: 'radii',
    borderBottomRightRadius: 'radii',
    borderBottomLeftRadius: 'radii',
  },
  selectors: {
    hover: SELECTOR_HORVER,
  },
})

describe('base styles', () => {
  const base = {
    px: '1',
    color: 'white',
    borderWidth: 'sm',
    borderRadius: 'sm',
    backgroundColor: 'colorPalette',
    colorPalette: 'blue.100',
    _hover: {
      color: 'yellow',
    },
  }

  const variants = {
    size: {
      sm: { w: '1', h: '1' },
      md: { w: '2', h: '2' },
    },
    color: {
      red: {
        color: 'red',
      },
      blue: {
        color: 'blue',
      },
    },
  }

  const baseStyles = defineRecipe({
    base,
    variants,
    compoundVariants: [
      {
        size: ['sm', 'md'],
        color: 'red',
        css: {
          background: 'red',
        },
      },
    ],
  })

  const { css } = createSystem(sysConfig)

  it('should defineRecipe work correctly', () => {
    expect(baseStyles()).toEqual(base)

    expect(
      baseStyles({
        size: 'sm',
      }),
    ).toEqual({
      ...base,
      ...variants.size.sm,
    })

    expect(
      baseStyles({
        size: 'sm',
        color: 'blue',
      }),
    ).toEqual({
      ...base,
      ...variants.size.sm,
      ...variants.color.blue,
    })

    expect(
      baseStyles({
        size: 'sm',
        color: 'red',
      }),
    ).toEqual({
      ...base,
      ...variants.size.sm,
      ...variants.color.red,
      background: 'red',
    })

    expect(
      baseStyles({
        size: 'md',
        color: 'red',
      }),
    ).toEqual({
      ...base,
      ...variants.size.md,
      ...variants.color.red,
      background: 'red',
    })
  })

  it('should css work correctly', () => {
    const toCssVariable = (v: string) => `var(--${CSS_VARS_PREFIX}-${v})`

    const transformedBase = {
      paddingLeft: toCssVariable('spaces-1'),
      paddingRight: toCssVariable('spaces-1'),
      color: toCssVariable('colors-white'),
      backgroundColor: toCssVariable('colors-blue-100'),
      borderWidth: toCssVariable('borders-sm'),
      borderRadius: toCssVariable('radii-sm'),
      [SELECTOR_HORVER]: {
        color: 'yellow',
      },
    }

    const transformedVariants = {
      size: {
        sm: {
          width: toCssVariable('sizes-1'),
          height: toCssVariable('sizes-1'),
        },
        md: {
          width: toCssVariable('sizes-2'),
          height: toCssVariable('sizes-2'),
        },
      },
      color: {
        red: {
          color: 'red',
        },
        blue: {
          color: 'blue',
        },
      },
    }

    expect(css(baseStyles())).toEqual(transformedBase)
    expect(
      css(
        baseStyles({
          size: 'sm',
        }),
      ),
    ).toEqual({
      ...transformedBase,
      ...transformedVariants.size.sm,
    })

    expect(
      css(
        baseStyles({
          size: 'sm',
          color: 'blue',
        }),
      ),
    ).toEqual({
      ...transformedBase,
      ...transformedVariants.size.sm,
      ...transformedVariants.color.blue,
    })

    expect(
      css(
        baseStyles({
          size: 'sm',
          color: 'red',
        }),
      ),
    ).toEqual({
      ...transformedBase,
      ...transformedVariants.size.sm,
      ...transformedVariants.color.red,
      background: 'red',
    })

    expect(
      css(
        baseStyles({
          size: 'md',
          color: 'red',
        }),
      ),
    ).toEqual({
      ...transformedBase,
      ...transformedVariants.size.md,
      ...transformedVariants.color.red,
      background: 'red',
    })
  })
})
