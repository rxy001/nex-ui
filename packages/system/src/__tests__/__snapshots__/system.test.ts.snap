// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css should convert tokens into corresponding CSS variables 1`] = `
{
  "backgroundColor": "var(--test-colors-primary)",
  "borderColor": "var(--test-colors-blue-100)",
  "borderRadius": "var(--test-radii-sm)",
  "borderWidth": "var(--test-border-widths-sm)",
  "boxShadow": "var(--test-shadows-sm)",
  "color": "var(--test-colors-white)",
  "fontFamily": "var(--test-font-families-body)",
  "fontSize": "var(--test-font-sizes-sm)",
  "fontWeight": "var(--test-font-weights-medium)",
  "height": "var(--test-sizes-2)",
  "lineHeight": "var(--test-line-heights-base)",
  "margin": "var(--test-spaces-1)",
  "padding": "var(--test-spaces-1)",
  "transition": "var(--test-transitions-all)",
  "width": "var(--test-sizes-1)",
  "zIndex": "var(--test-z-indices-min)",
}
`;

exports[`css should convert tokens into corresponding CSS variables 2`] = `
Map {
  "base" => Map {
    "--test-colors-white" => "#f1f1f1",
    "--test-colors-blue-100" => "#dbeafe",
    "--test-colors-blue-200" => "#bfdbfe",
    "--test-colors-blue-300" => "#a3cfff",
    "--test-colors-red-100" => "#fca5a5",
    "--test-borders-sm" => "1px solid",
    "--test-radii-sm" => "4px",
    "--test-radii-md" => "6px",
    "--test-radii-lg" => "8px",
    "--test-sizes-1" => "32px",
    "--test-sizes-2" => "40px",
    "--test-sizes-3" => "48px",
    "--test-sizes-1-5" => "36px",
    "--test-spaces-1" => "4px",
    "--test-spaces-2" => "8px",
    "--test-spaces-3" => "12px",
    "--test-font-sizes-sm" => "12px",
    "--test-font-sizes-md" => "14px",
    "--test-font-sizes-lg" => "16px",
    "--test-font-weights-normal" => 400,
    "--test-font-weights-medium" => 500,
    "--test-font-weights-semibold" => 600,
    "--test-line-heights-short" => 1.375,
    "--test-line-heights-base" => 1.5,
    "--test-line-heights-tall" => 1.625,
    "--test-font-families-body" => "system",
    "--test-z-indices-min" => 1,
    "--test-transitions-all" => "all 0.2s",
    "--test-shadows-sm" => "shadow",
    "--test-border-widths-sm" => "1px",
    "--test-colors-primary" => "var(--test-colors-blue-100)",
    "--test-colors-secondary" => "var(--test-colors-blue-200)",
  },
  "dark" => Map {
    "--test-colors-primary" => "var(--test-colors-blue-300)",
  },
}
`;
